src/modules/social/sagas.ts

```diff
@@ -166,4 +172,49 @@ export function* socialSagas() {
       yield put(rejectFriendshipFailure(isErrorWithMessage(error) ? error.message : 'Unknown'))
     }
   }
+
+  function* handleLogInAndRequestFriendshipRequest(action: LogInAndRequestFriendshipRequestAction) {
+    const friendAddress = action.payload
+
+    yield put(openModal('LoginModal'))
+
+    const {
+      success,
+      close
+    }: {
+      success: ConnectWalletSuccessAction
+      close: CloseModalAction
+    } = yield race({
+      success: take(CONNECT_WALLET_SUCCESS),
+      close: take(CLOSE_MODAL)
+    })
+
+    if (close || (success && success.payload.wallet.address.toLowerCase() === friendAddress)) {
+      return
+    }
+
+    const { failureInitializingSocialClient } = yield race({
```

Comment:
When are the actions from the races called? Can we expect a race condition and the saga never to finish?

Reply:
These actions are related to the log in, the initialization of the social client and the friends request fetching. We need to wait for all of them to be completed to ensure the proper termination of the saga. The usual flow goes as follow:
- The user logs in successfully and the saga triggers the initialization of the social client.
- The social client is successfully initialized, triggering the fetch of friends.
To prevent any possible issues and be sure that the saga will work as intended, we need to:
- Wait for the user to log in successfully or close the modal. If the log in is not successful, the user remains in the modal until they close it. If the user closes the modal, the saga will finish.
- Wait for the initialization of the client. If it's unsuccessful, the saga will finish.
- Wait for the fetching of friends. If it's unsuccessful, the saga will finish.